sky/spot/controller.py

```diff
@@ -74,8 +75,48 @@ def __init__(self, job_id: int, dag_yaml: str,
                 job_id_env_vars)
             task.update_envs(task_envs)
 
+    def _download_log_and_stream(
+            self,
+            handle: cloud_vm_ray_backend.CloudVmRayResourceHandle) -> None:
+        """ Downloads the latest log from spot cluster, and stream them
+
+        We do not stream the logs from the spot cluster directly, as the
+        donwload and stream should be faster, and more robust against
+        preemptions or ssh disconnection during the streaming.
+        """
+        spot_job_logs_dir = os.path.join(constants.SKY_LOGS_DIRECTORY,
+                                         'spot_jobs')
+        os.makedirs(spot_job_logs_dir, exist_ok=True)
+        try:
+            log_dirs = self._backend.sync_down_logs(
+                handle,
+                # Download the log for the latest job.
+                job_ids=None,
```

Comment:
I was confused here for a sec. For now, each spot cluster can only have job id 0 (or is it 1), right? Can we explicitly specify that on both L93 ("latest" is confusing) and L94?

Reply:
No, the job id can be larger than 1, because for the multi-node case, it is possible we recover the job on an existing cluster with only the worker node preempted, which will lead to a job_id larger than 1. It should be more robust to get the log for the latest job.

Reply:
Ah I see. Worth copying this into a comment :) All these "job ids" correspond to the same logical spot job self._job_id, right?

Reply:
Good point! Let me copy it. Yes, all the job_ids correspond to the same logical spot job.